releases/stable-branches-and-release-cycle.md

```diff
@@ -0,0 +1,79 @@
+# Producing and supporting stable releases
+
+## Summary
+
+Today the Submariner project releases one minor version a month without capacity
+for stable branches or bugfix releases. In the following proposal we define a plan
+to make this possible starting with 0.8.
+
+## Proposal
+
+### Alignment of branch -> image releases
+
+We produce currently `<image>:devel` for every master push, along with `<image>:x.y.z` for
+the releases.
+
+Changing the `master` branch name to `devel` would make this homogeneous so our image releases
+are simply `<image>:<branch-or-tag>`.
+
+### Creating one stable branch for every minor release
+
+Approaching the release of every minor or major version a new stable branch will be forked
+off `devel` branch in the form of `release-x.y` for every existing repository.
+
+Any changes related to pinning shipyard base image will be done in that branch.
+
+At that point in time, any patches which belong to the release will need to be merged in master
+and backported to `release-x.y`. For this reason, the later we branch, the less extra work
+will be necessary to produce a stable release.  
+
+### Maintaining a stable branch
+
+Any patches that are necessary for fixing bugs in a stable branch will need to be merged
+first into master, and the later stable branches.
+
+`release-0.8`  <-- `release-0.9` <-- `devel`
+
+Patches could be simple cherry-picks, or sometimes they could need specific adaptations
+to the branch.
+
+When a patch is merged to a stable branch the images for such branch are updated, for example
+`quay.io/submariner/submariner:release-0.9`
```

Comment:
Instead of updating release-0.9 image itself, we should probably create a new image like `quay.io/submariner/submariner:release-0.9.x" and increment x as we go along. WDYS?

Reply:
I agree as we do not expect such changes so often
so a patch would increment the z-stream number

Reply:
I understand the `release-0.9` images as being “branch” images, _i.e._ updated to always point to the tip of the corresponding branch. _Release_ images would be tagged as they are currently, 0.9.0 etc.

Reply:
No, it should not do that.

The for doing that we should make a release explicitly, it would be a stable release 0.9.1 once we consider that release-0.9 has all the right patches.

Reply:
The problem is that sometimes a new stable version will need multiple patches. Or sometimes we will want to control the cadence at which new patch releases are created.

Reply:
> I understand the `release-0.9` images as being “branch” images, _i.e._ updated to always point to the tip of the corresponding branch. _Release_ images would be tagged as they are currently, 0.9.0 etc.

Yes that's the gist, the `release-0.9` image in this example is the "latest stable" but not a "released" image, the "released" image will be tagged as always e.g. `0.9.3` or whatever the version will be.